File: content/search/source/includes/cross-collection-tutorial/code-snippets/csharp/CreateIndexExample-materialized.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Replace the connection string with your MongoDB deployment's connection string
var uri = "<connection-string>";

var client = new MongoClient(uri);

var db = client.GetDatabase("sample_supplies");
var collection = db.GetCollection<BsonDocument>("monthlyPhoneTransactions");

// Create the MongoDB Search index definition
var index =  new CreateSearchIndexModel(
  "monthlySalesIndex", new BsonDocument
  {
    { "mappings", new BsonDocument
      {
        { "dynamic", true}
      }
    }
  });

var result = collection.SearchIndexes.CreateOne(index);
Console.WriteLine($"New index name: {result}");

File: content/search/source/includes/cross-collection-tutorial/code-snippets/nodejs/create-index-example-materialized.js
const { MongoClient } = require("mongodb");

// Replace the connection string with your MongoDB deployment's connection string
const uri =  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_supplies");
    const collection = database.collection("monthlyPhoneTransactions");

    // Create the MongoDB Search index definition
    const index = {
        name: "monthlySalesIndex",
        definition: {
            "mappings": {
                "dynamic": true
            }
        }
    }

    // Create the index
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}

run().catch(console.dir);
